examples/Billing/AddBillingSetup.cs

```diff
@@ -0,0 +1,224 @@
+        /// <param name="paymentsAccountId">Optional payments account ID to attach to the new
+        ///     billing setup. Must be formatted as "1234-5678-9012-3456".</param>
+        /// <param name="paymentsProfileId">Optional payments profile ID to attach to a new payments
+        ///     account and to the new billing setup. Must be formatted as "1234-5678-9012".</param>
+        /// <returns>A new BillingSetup instance with complete payment details.</returns>
+        /// <exception cref="Exception">Generic exception if no payment details have been
+        ///     provided.</exception>
+        private BillingSetup CreateBillingSetup(long customerId, string paymentsAccountId,
+            string paymentsProfileId)
+        {
+            BillingSetup billingSetup = new BillingSetup();
+
+            // Set the appropriate payments account field.
+            if (paymentsAccountId != null)
+            {
+                // If a payments account id has been provided, set PaymentsAccount to its resource
+                // name. You can list available payment accounts via the PaymentAccountService's
+                // ListPaymentsAccounts method.
+                billingSetup.PaymentsAccount =
+                    ResourceNames.PaymentsAccount(customerId, paymentsAccountId);
+            }
+            else if (paymentsProfileId != null)
+            {
+                // Otherwise, create a new payments account by setting the PaymentsAccountInfo
+                // field. See https://support.google.com/google-ads/answer/7268503 for information
+                // about payments profiles.
+                billingSetup.PaymentsAccountInfo = new BillingSetup.Types.PaymentsAccountInfo()
+                {
+                    PaymentsAccountName = "Payments Account #" + ExampleUtilities.GetRandomString(),
+                    PaymentsProfileId = paymentsProfileId
+                };
+            }
+            else
+            {
+                throw new Exception("No paymentsAccountId or paymentsProfileId provided.");
+            }
+
+            return billingSetup;
+        }
+
+        /// <summary>
+        /// Sets the starting date time for the new billing setup. Queries the customer's account
+        /// to see if there are any approved billing setups. If there are any, the new billing setup
+        /// may have a starting date time of any day in the future. If not, the billing setup must
+        /// be set to start immediately.
+        /// </summary>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="googleAdsService">The Google Ads service client.</param>
+        /// <param name="billingSetup">The instance of BillingSetup whose starting date time will
+        ///     be set.</param>
+        private void SetBillingSetupStartDateTime(long customerId, GoogleAdsServiceClient
```

Comment:
I think you should set the end date too (1 day after the start date) so the code example can be run multiple times without reaching the indefinite ending date case.

Reply:
Done.